cmake/CeleritasAddTest.cmake

```diff
@@ -309,13 +309,16 @@ function(celeritas_add_test SOURCE_FILE)
     # Create an executable and link libraries against it
     add_executable(${_TARGET} "${SOURCE_FILE}" ${PARSE_SOURCES})
 
-    celeritas_target_link_libraries(${_TARGET} Celeritas::Core)
+    # Note: for static linking the library order is relevant.
 
     celeritas_target_link_libraries(${_TARGET}
-      Celeritas::Test
       ${CELERITASTEST_LINK_LIBRARIES}
       ${PARSE_LINK_LIBRARIES}
+      Celeritas::Test
     )
+
+    celeritas_target_link_libraries(${_TARGET} Celeritas::Core)
```

Comment:
You should be able to move this into the `celeritas_target_link_libraries` above